src/server/scripts/Outland/HellfireCitadel/MagtheridonsLair/boss_magtheridon.cpp

```diff
@@ -16,617 +15,586 @@
+                    DoCastAOE(SPELL_CAMERA_SHAKE);
+                    events.ScheduleEvent(EVENT_COLLAPSE, Seconds(6));
+                }
             }
-            void MoveInLineOfSight(Unit* who) override
 
+            void JustDied(Unit* /*killer*/) override
             {
-                if (!trigger)
-                    ScriptedAI::MoveInLineOfSight(who);
+                Talk(SAY_DEATH);
+                _JustDied();
+                instance->SetData(DATA_MANTICRON_CUBE, ACTION_DISABLE);
+            }
+
+            void KilledUnit(Unit* who) override
+            {
+                if (who->GetTypeId() == TYPEID_PLAYER)
+                    Talk(SAY_SLAY);
             }
 
             void UpdateAI(uint32 diff) override
             {
-                if (trigger)
-                {
-                    if (trigger == 1)
-                    {
-                        if (FireBlast_Timer <= diff)
-                        {
-                            DoCast(me, SPELL_DEBRIS_DAMAGE, true);
-                            trigger = 3;
-                        }
-                        else FireBlast_Timer -= diff;
-                    }
+                if (!events.IsInPhase(PHASE_COLLAPSE) && !UpdateVictim())
                     return;
-                }
 
-                if (Despawn_Timer <= diff)
-                {
-                    me->DespawnOrUnsummon();
-                }
-                else Despawn_Timer -= diff;
-
-                if (!UpdateVictim())
+                if (me->HasUnitState(UNIT_STATE_CASTING))
                     return;
 
-                if (FireBlast_Timer <= diff)
+                events.Update(diff);
+
+                while (uint32 eventId = events.ExecuteEvent())
                 {
-                    DoCastVictim(SPELL_FIRE_BLAST);
-                    FireBlast_Timer = 5000 + rand32() % 10000;
+                    switch (eventId)
+                    {
+                        case EVENT_BERSERK:
+                            DoCast(SPELL_BERSERK);
+                            break;
+                        case EVENT_CLEAVE:
+                            DoCastVictim(SPELL_CLEAVE);
+                            events.Repeat(Seconds(10));
+                            break;
+                        case EVENT_BLAZE:
+                            me->CastCustomSpell(SPELL_BLAZE_TARGET, SPELLVALUE_MAX_TARGETS, 1);
+                            events.Repeat(Seconds(20));
+                            break;
+                        case EVENT_QUAKE:
+                            me->CastCustomSpell(SPELL_QUAKE, SPELLVALUE_MAX_TARGETS, 5);
+                            events.Repeat(Seconds(60));
+                            break;
+                        case EVENT_START_FIGHT:
+                            CombatStart();
+                            break;
+                        case EVENT_RELEASED:
+                            Talk(SAY_AGRO);
```

Comment:
Ok, but as I said above - don't know where this text should be used. Cannot find any movie with this text. Also can't find this text in broadcast_text table.
Maybe something can be found in sniffs.